edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/ConnectionHandler.cs

```diff
@@ -154,6 +154,31 @@ public async Task RegisterLinkHandler(ILinkHandler linkHandler)
             }
         }
 
+        public async Task RemoveLinkHandler(ILinkHandler linkHandler)
+        {
+            Preconditions.CheckNotNull(linkHandler);
+            using (await this.registryUpdateLock.LockAsync())
+            {
+                if (this.registry.ContainsKey(linkHandler.Type))
+                {
+                    this.registry.Remove(linkHandler.Type);
+                    if (this.registry.Count == 0)
+                    {
+                        await this.CloseConnection();
+                    }
+                }
+            }
+        }
+
+        async Task CloseConnection()
+        {
+            using (await this.initializationLock.LockAsync())
```

Comment:
There's two locks in play here. Is there any chance for a deadlock?

Reply:
No, because the 2 locks are not used together in the other case.